File: app/src/main/java/orinasa/njarasoa/maripanatokana/domain/model/WeatherData.kt
package orinasa.njarasoa.maripanatokana.domain.model

data class WeatherData(
    val temperature: Temperature,
    val feelsLike: Temperature,
    val tempMin: Temperature,
    val tempMax: Temperature,
    val weatherCode: Int,
    val iconCode: String,
    val locationName: String,
    val pressure: Pressure,
    val humidity: Int, // percentage
    val dewPoint: Temperature,
    val windSpeed: WindSpeed,
    val windDeg: Int, // degrees
    val windGust: WindSpeed?,
    val rain: Precipitation?,
    val snow: Precipitation?,
    val uvIndex: Double,
    val visibility: Int, // meters
    val sunrise: Long, // epoch seconds
    val sunset: Long, // epoch seconds
    val dailySunrise: List<Long> = emptyList(), // epoch millis per day
    val dailySunset: List<Long> = emptyList(), // epoch millis per day
    val hourlyForecast: List<HourlyForecast> = emptyList(),
    val dailyForecast: List<DailyForecast> = emptyList(),
    val timestamp: Long = System.currentTimeMillis(),
)

File: react-native/src/domain/WeatherData.ts
import {Temperature} from './Temperature';
import {WindSpeed} from './WindSpeed';
import {Pressure} from './Pressure';
import {Precipitation} from './Precipitation';

export interface HourlyForecast {
  time: number; // epoch millis
  temperature: Temperature;
  weatherCode: number;
  precipProbability: number;
}

export interface DailyForecast {
  date: number; // epoch millis
  tempMax: Temperature;
  tempMin: Temperature;
  weatherCode: number;
  precipProbability: number;
}

export interface WeatherData {
  temperature: Temperature;
  feelsLike: Temperature;
  tempMin: Temperature;
  tempMax: Temperature;
  weatherCode: number;
  iconCode: string;
  locationName: string;
  pressure: Pressure;
  humidity: number; // percentage
  dewPoint: Temperature;
  windSpeed: WindSpeed;
  windDeg: number; // degrees
  windGust: WindSpeed | null;
  rain: Precipitation | null;
  snow: Precipitation | null;
  cloudCover: number; // percentage
  uvIndex: number;
  visibility: number; // meters
  sunrise: number; // epoch seconds
  sunset: number; // epoch seconds
  dailySunrise: number[]; // epoch millis per day
  dailySunset: number[]; // epoch millis per day
  hourlyForecast: HourlyForecast[];
  dailyForecast: DailyForecast[];
  timestamp: number; // millis
}
